test/test.ml

```diff
@@ -396,32 +391,35 @@ module Make(IO : Redis.S.IO) = struct
       (fun scanned_fields ->
          List.length fields = List.length scanned_fields)
 
+  let cleanup_keys conn =
+    Client.keys conn "ounit_*" >>= Client.del conn
 
   let bracket test_case () =
-    IO.run
-      (setup () >>= fun conn ->
-       test_case conn >>= fun () ->
-       teardown conn)
+    IO.run @@ Client.with_connection redis_spec test_case
```

Comment:
If the code uses `@@` the OCaml version should be restricted to >= 4.01.0 in `_oasis`.  That should also simplify the code generated by oasis.